account_members.go

```diff
@@ -81,18 +83,73 @@ func (api *API) AccountMembers(ctx context.Context, accountID string, pageOpts P
 //
 // API reference: https://api.cloudflare.com/#account-members-add-member
 func (api *API) CreateAccountMemberWithStatus(ctx context.Context, accountID string, emailAddress string, roles []string, status string) (AccountMember, error) {
+	invite := AccountMemberInvitation{
+		Email:    emailAddress,
+		Roles:    roles,
+		Policies: nil,
+		Status:   status,
+	}
+	return api.CreateAccountMemberInternal(ctx, accountID, invite)
+}
+
+// CreateAccountMember invites a new member to join an account with roles.
+// The member will be placed into "pending" status and receive an email confirmation.
+// NOTE: If you are currently enrolled in Domain Scoped Roles, your roles will be converted to policies
+// upon member invitation. We recommend upgrading to CreateAccountMemberWithPolicies to use policies.
+//
+// API reference: https://api.cloudflare.com/#account-members-add-member
+func (api *API) CreateAccountMember(ctx context.Context, accountID string, emailAddress string, roles []string) (AccountMember, error) {
+	invite := AccountMemberInvitation{
+		Email:    emailAddress,
+		Roles:    roles,
+		Policies: nil,
+		Status:   "",
+	}
+	return api.CreateAccountMemberInternal(ctx, accountID, invite)
+}
+
+// CreateAccountMemberWithRoles is a terse wrapper around the CreateAccountMember method
+// for clarity on what permissions you're granting an AccountMember.
+//
+// API reference: https://api.cloudflare.com/#account-members-add-member
+func (api *API) CreateAccountMemberWithRoles(ctx context.Context, accountID string, emailAddress string, roles []string) (AccountMember, error) {
+	return api.CreateAccountMember(ctx, accountID, emailAddress, roles)
+}
+
+// CreateAccountMemberWithPolicies invites a new member to join your account with policies.
+// Policies are the replacement to legacy "roles", which enables the newest feature Domain Scoped Roles.
+//
+// API documentation will be coming shortly. Blog post: https://blog.cloudflare.com/domain-scoped-roles-ga/
+func (api *API) CreateAccountMemberWithPolicies(ctx context.Context, accountID string, emailAddress string, policies []Policy) (AccountMember, error) {
+	invite := AccountMemberInvitation{
+		Email:    emailAddress,
+		Roles:    nil,
+		Policies: policies,
+		Status:   "",
+	}
+	return api.CreateAccountMemberInternal(ctx, accountID, invite)
+}
+
+// CreateAccountMemberInternal allows you to provide a raw AccountMemberInvitation to be processed
+// and contains the logic for other CreateAccountMember* methods
+func (api *API) CreateAccountMemberInternal(ctx context.Context, accountID string, invite AccountMemberInvitation) (AccountMember, error) {
+	// make sure we have account
```

Comment:
let's remove comments like this one. i find it more of a smell since if we need these style of comments, the code isn't clear enough. a good overview of where we aim to go with code comments in this repository: https://stackoverflow.blog/2021/12/23/best-practices-for-writing-code-comments/